include/dgl/runtime/util.h

```diff
@@ -8,6 +8,14 @@
 
 #include "c_runtime_api.h"
 
+#ifdef _MSC_VER
+// rand in MS compiler works well in multi-threading.
+static inline int rand_r(unsigned *seed) {
+  return rand();
+}
+#define _CRT_RAND_S
+#endif
+
```

Comment:
I'm not sure how to use `std::uniform_int_distribution` properly for this purpose since the range of random number is always changing.  It seems that they want us to initialize a different RNG every time the lower/upper bound changes?  Here's the reference code:
```C++
    std::random_device rd;  //Will be used to obtain a seed for the random number engine
    std::mt19937 gen(rd()); //Standard mersenne_twister_engine seeded with rd()
    std::uniform_int_distribution<> dis(1, 6);
 
    for (int n=0; n<10; ++n)
        //Use dis to transform the random unsigned int generated by gen into an int in [1, 6]
        std::cout << dis(gen) << ' ';
```

Reply:
According to [cppreference](http://www.cplusplus.com/reference/random/), generator and distribution are two things.
* Generator for generating random numbers from the seed.
* Distribution for converting random numbers to certain distribution.
In our case, I think each call will create a new distribution object that uses the same generator object.

For multi-thread case, according to [this stackoverflow](https://stackoverflow.com/questions/24007263/is-random-number-generation-in-multiple-threads-using-c11-random-library-slow), we should create one generator for each thread so there is no interlock between threads. There will be only one `random_device` and locking is required to initialize the generator but the overhead is one-time.

We need to write benchmark code for this.